.github/workflows/publish-prerelease.yml

```
@@ -0,0 +1,70 @@
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  publish-prerelease:
+    if: ${{ github.ref != 'refs/heads/main' }}
+    runs-on: ubuntu-latest
+    env:
+      CHANNEL: ${{ github.event_name == 'workflow_dispatch' && github.event.inputs.channel || (startsWith(github.ref_name, 'prerelease/alpha/') && 'alpha' || 'beta') }}
+    steps:
+      - uses: actions/checkout@v3
```

actionlint (1.7.12)

[error] the runner of "actions/checkout@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)